File: src/main/java/io/xol/chunkstories/api/particles/ParticlesManager.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.api.particles;

import org.joml.Vector3dc;

public interface ParticlesManager
{
	public void spawnParticleAtPosition(String particleTypeName, Vector3dc position);
	
	public void spawnParticleAtPositionWithVelocity(String particleTypeName, Vector3dc position, Vector3dc velocity);
}

File: src/main/java/xyz/chunkstories/api/particles/ParticlesManager.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.particles

import org.joml.Vector3dc

interface ParticlesManager {
    //fun spawnParticleAtPosition(particleTypeName: String, position: Vector3dc)
    //fun spawnParticleAtPositionWithVelocity(particleTypeName: String, position: Vector3dc, velocity: Vector3dc)
    fun <T: ParticleType.Particle> spawnParticle(typeName: String, init: T.() -> Unit)

    fun <T: ParticleType.Particle> spawnParticle(type: ParticleTypeDefinition, init: T.() -> Unit)
}
